### PR TITLE
Ast io refactor

### PR DESCRIPTION
**Pull request by yuli6313**
Remove stores and loads from AstRelation and remove the dependencies. Move deprecated io warning to parser.yy. The tests still need to be changed.

### CODE REVIEW

**Comment by b-scholz** [comment]
You need to update the error messages in the tests syntactic/input, syntactic/output/, and qualifiers2.

File: src/MagicSet.cpp [hunk 1]
```diff
@@ -1161,7 +1161,7 @@ bool MagicSetTransformer::transform(AstTranslationUnit& translationUnit) {
                         newDirective->addKVP("filename", originalName.getNames()[0] + ".facts");
                     }
 
-                    newRelation->addLoad(std::unique_ptr<AstLoad>(newDirective));
+                    program->addLoad(std::unique_ptr<AstLoad>(newDirective));
```
Comment (mmcgr) [inline src/MagicSet.cpp]: I'm not sure that this is needed, or if anything in MagicSet related to IO is needed.

File: src/parser.yy [hunk 2]
```diff
@@ -279,6 +279,24 @@ unit
     }
   | unit relation_decl {
         for (auto* cur : $relation_decl) {
+            if ((cur->getQualifier() & INPUT_RELATION) != 0) {
+                auto load = std::make_unique<AstLoad>();
+                load->addName(cur->getName());
+                load->setSrcLoc(cur->getSrcLoc());
+                driver.addLoad(std::move(load));
```
Comment (mmcgr) [inline src/parser.yy]: I don't think the std::move is needed, here and below. If the compiler doesn't object, remove it.

Comment (yuli6313) [reply src/parser.yy]: This cannot to be moved, otherwise it does not compile.

Comment (mmcgr) [reply src/parser.yy]: Ah, that actually makes sense - if it was being returned it would be automatically moved, but that's not the case here.

**Comment by mmcgr** [comment]
The sections of MagicSet related to AstStores can be removed, since the relation name does not change, and they are not stored within the removed relations.

The sections of MagicSet related to AstLoads can be simplified to just update the name.

The syntactic/printsize test still has incorrect output

File: src/ComponentInstantiationTransformer.cpp [hunk 3]
```diff
@@ -508,37 +508,37 @@ bool ComponentInstantiationTransformer::transform(AstTranslationUnit& translatio
     // remember the remaining orphan clauses
     program.clauses.clear();
     program.clauses.swap(unbound);
+    /*
```
Comment (azreika) [inline src/ComponentInstantiationTransformer.cpp]: Why is this here but commented out? Can this be removed? Or is there an issue with the current implementation that needs this as a reference?

Comment (yuli6313) [reply src/ComponentInstantiationTransformer.cpp]: It should be removed. AstRelations doesn't store related AstStore or AstLoad any more.

Comment (azreika) [reply src/ComponentInstantiationTransformer.cpp]: OK, I will remove it in my next PR.
